### ISSUE TITLE
List of known program conflicts with MacType.

### ISSUE THREAD

**Issue by sammilucia** [snowie2000/mactype#426]
A place to collect all known conflicts, e.g:
- EXE won't start
- BSOD / GSOD / CTD
- EXE starts but errors
- Completely unreadable/unusable

This is NOT for software which is working, but with minor font rendering defects.
- AoEDE.exe (Age of Empires - Definitive Edition, "rendering failed, displayed strangely" from #406)

**Comment by ShintakuNobuhiro** [comment]
Cannot Start VM with Oracle VirtualBox

**Comment by Mahayu** [comment]
Touhou series should't use Mactype or GDI++, as it will make subtitles coverd by black lanes.

**Comment by snowie2000** [comment]
@ShintakuNobuhiro VM is known to be incompatible with any dll injection.
You have to add VirtualBox.exe to unloaddll section to avoid it.

**Comment by snowie2000** [comment]
UWP is always a mystery. No one knows what happened, even those Microsoft guys.

**Comment by sylveon** [comment]
It's not really a UWP only problem: as noted later in the thread, the desktop build experiences the same issue. Furthermore, the version distributed on the Store is just a Win32 app packaged with the Desktop Bridge (which creates an .appx file holding conventional Win32 programs).

**Comment by snowie2000** [comment]
Random crashes are really annoying for debugging.
Even it's just a wrap for Win32 app, it runs under the UWP isolated environment which is known to be affected by MacType.

**Comment by snowie2000** [comment]
@sylveon Can you do a mini dump when this app crashes?

**Comment by sylveon** [comment]
A memory dump from the desktop build (so UWP shouldn't be messing stuff) was provided in the issue, you can use it to debug.

**Comment by snowie2000** [comment]
Good, I see it. Thanks.

**Comment by snowie2000** [comment]
@sylveon I justed the code a little and merged some of the assemblers from the latest Easyhook.
Let's see if it make the crash situation any better.

[easyhook-0918.zip](https://github.com/snowie2000/mactype/files/2391077/easyhook-0918.zip)

**Comment by sammilucia** [comment]
@snowie2000 should this be used with the 20180917 beta or it doesn't matter?

**Comment by snowie2000** [comment]
It's version independent.

**Comment by snowie2000** [comment]
@sylveon @sammilucia I found another problem that in some rare cases, the hook structure allocated will be freed twice. That may be the real case. Try it.
[easyhook-0918-2.zip](https://github.com/snowie2000/mactype/files/2391406/easyhook-0918-2.zip)

**Comment by snowie2000** [comment]
[easyhook-20180918-2.zip](https://github.com/snowie2000/mactype/files/2391713/easyhook-20180918-2.zip)
Try this one instead although I believe the previous one has already fixed the problem. (I re-fixed a problem that's not exist)

**Comment by sylveon** [comment]
I don't have a VM to test this on, but I'll set one up when I have time (I wish that enabling Hyper-V on my machine doesn't makes it get stuck on the spinning rings)

**Comment by sylveon** [comment]
Ok, after installing MacType in a VM, I first had issues with opening TranslucentTB's context menu, and then with opening the color picker. Never crashing at startup (unlike reported in my issue tracker).

After replacing the EasyHook DLLs, both issues where fixed.

**Comment by snowie2000** [comment]
And while there are no more crash problems with translucentTB, does MacType still have the effect on it?

**Comment by sylveon** [comment]
Yes, it still applies correctly. I can't do much playing around, as TranslucentTB's interface is basically just a context menu.

**Comment by snowie2000** [comment]
That's interesting as the code indicates that the problem should only occur when the hook fails. :thinking: 
So you won't get locked in the Windows 10 logon screen anymore. Are you using the latest beta with just the public version I released last year with patched dlls?

**Comment by sylveon** [comment]
I'm using the latest Skip Ahead build, but I only use MacType (with the EasyHook dll updated by the ones on https://github.com/snowie2000/mactype/issues/426#issuecomment-422296797) by starting the tray app manually, so it doesn't applies to the lock screen.

**Comment by snowie2000** [comment]
Could you try the MacType beta version in the release section with the Registry mode?
Many thanks.

**Comment by sylveon** [comment]
I've extracted the beta over the latest release, replaced the EasyHook dlls with the ones in this thread, ran MacWiz, set it to Registry (and created+set a profile with red text to clearly see the difference) and restarted.

I noticed it wasn't working at all, so I disabled Secure Boot for the VM because the registry mode is probably using AppInit_DLL, which is not working with Secure Boot on. After that it worked.

So far so good, nothing seems to be crashing (including TranslucentTB and the lock screen) and the text stays in an obnoxious red no matter what I try.

![image](https://user-images.githubusercontent.com/6440374/45912025-393b0900-bde9-11e8-8fc4-af4ed967ee91.png)

**Comment by snowie2000** [comment]
Yes, secure boot must be disabled for registry mode to take effect or Windows will ignore the AppInit_DLL value.
The text color is really weird. Does the weird text color appear with the old MacType too?

**Comment by sammilucia** [comment]
Thanks @sylveon that reminded me - I've updated the Readme with the info about Secure Boot

**Comment by sammilucia** [comment]
@snowie2000 20180917 beta + easyhook-20180918-2 is working flawlessly for me on win8.1. I think we may have a new RC if no one has any problems. great work! 😊

**Comment by snowie2000** [comment]
> @snowie2000 20180917 beta + easyhook-20180918-2 is working flawlessly for me on win8.1. I think we may have a new RC if no one has any problems. great work!

Lock and load, ready to fire. :laughing:

**Comment by justablob** [comment]
Can't apply MacType to VS Code Insiders 1.28.0 and Discord (both use Electron). Im starting to think this is impossible to do, any updates on compatibility with Chrome/Chromium? Any secret flags?

**Comment by snowie2000** [comment]
You can enable DirectWrite basic support through directive, but it's still very primitive.

**Comment by Plaer1** [comment]
The new COD wont launch with MacType enabled lol.

**Comment by sammilucia** [comment]
@snowie2000 regarding some Java apps (IDEA, etc.) and some games e.g. CoD4 maybe we should develop a standard list of apps to be added to the exclusions, and clipboxfix.

I'm not talking about directwrite apps, just apps that don't work because they don't do fonts properly so MacType either needs to be disabled or the clipbox workaround.

**Comment by snowie2000** [comment]
@sammilucia Maybe we could build a "cloud" app list that needs special treatment. :smile:

**Comment by sammilucia** [comment]
haha

**Comment by guytanium** [comment]
Windows 10 built-in Mail app crashes either on loading or soon after with MacType enabled. What mode do I run it in to disable it for certain apps?

**Comment by snowie2000** [comment]
What you should do is to disable the directwrite support.

**Comment by snowie2000** [comment]
If you're still using the old version, not the beta 5 please upgrade

**Comment by guytanium** [comment]
> If you're still using the old version, not the beta 5 please upgrade

Sorry I was on the old version. Thankyou that fixed my issue.

**Comment by ShintakuNobuhiro** [comment]
Forza Horizon 4 can't launch with MacType beta-5 with Windows 10 Version 1809.
Fix：Change to standalone mode and exclude Forza Horizon 4 in the process manager.

(I'm Japanese. I used Google Translate. Sorry.)

**Comment by snowie2000** [comment]
If you can't start the game with it in the exclude list, please add it to the unloaddll list as MacType will still be injected into the process when it is "excluded"

**Comment by 114514hon** [comment]
> 
> 
> If you can't start the game with it in the exclude list, please add it to the unloaddll list as MacType will still be injected into the process when it is "excluded"

Add ForzaHorizon4.exe to the unloaddll list works in 2018_1019_beta4r2 with Windows 10 Version 1809, but not work in 2018.1-beta5. Forza Horizon 4 still can't launch after I add it to the unloaddll list.

I found a workaround, I add svchost.exe in unloaddll list and Forza Horizon 4 start.

**Comment by GNU-Linuxer** [comment]
MacType is conflict with VirtualBox (starting from 5.0), and you need to whitelist VirtualBox, otherwise every virtual machines won't start.

**Comment by Stzx** [comment]
In the `Windows 10 1809` release, the `MacType 2018.1-beta5` `EasyHK64.dll` module conflicts with Taskmgr.exe, the dump log shows attempting to read and write virtual addresses, but no permissions.

[Taskmgr.exe.12936.zip](https://github.com/snowie2000/mactype/files/3050248/Taskmgr.exe.12936.zip)

**Comment by snowie2000** [comment]
That will be very helpful if you can upload the dump log.

**Comment by w-biggs** [comment]
This seems to occur with Ableton Live 10's `Ableton Index.exe` and `Ableton Web Connector.exe` with the newest version (`2018.1-beta5`). I uploaded the minidumps to Mega (too big for github, over 10MB):

https://mega.nz/#!P1YjkKbZ!dgIZG8lhZpVWsQZ3uFuYP5C9Gn3tr4EEuYtxLpNzdCQ

Adding them to the `UnloadDll` section of `MacType.ini` doesn't fix the issue.

**Comment by sammilucia** [comment]
Hi @w-biggs, could you please also try the latest Live beta (10.1b18) as they've made a lot of fixes to the font rendering and I have not had this problem since updating.

You can sign up here https://www.ableton.com/en/beta/.. You can install the beta separately if you depend on Live for production.

Thank you

**Comment by w-biggs** [comment]
I actually already am a part of the beta, I just didn't have it installed - and it works! Odd. It's probably still worth looking at why the crashing occurred, but this definitely works as a workaround.

**Comment by snowie2000** [comment]
> This seems to occur with Ableton Live 10's `Ableton Index.exe` and `Ableton Web Connector.exe` with the newest version (`2018.1-beta5`). I uploaded the minidumps to Mega (too big for github, over 10MB):
> 
> https://mega.nz/#!P1YjkKbZ!dgIZG8lhZpVWsQZ3uFuYP5C9Gn3tr4EEuYtxLpNzdCQ
> 
> Adding them to the `UnloadDll` section of `MacType.ini` doesn't fix the issue.

Thank you for mini dump.
As we can see from your dump, none of the MacType component (core, easyhook) had been loaded into the ableton indexer and connector. So I really have no idea why it didn't start.

![QQ截图20190409111928](https://user-images.githubusercontent.com/11767189/55771154-6dda8d80-5ab9-11e9-9ed2-bdb11926ab3a.png)
![QQ截图20190409111951](https://user-images.githubusercontent.com/11767189/55771155-6dda8d80-5ab9-11e9-9ff9-33829b5210c7.png)

**Comment by snowie2000** [comment]
There is however another possibility that the problem was about the program which spawned the indexer.
Try to disable the option `HookChildProcesses` and remove abelton from the unloaddll section, and try again to see if the indexer could back to work.

**Comment by w-biggs** [comment]
`HookChildProcesses=0` didn't help fix the issue either unfortunately. From what I can tell from analyzing the minidumps (I'm not an expert at this so I could be misinterpreting it) MacType seems to be using or overwriting some memory that Ableton is trying to access, which is throwing the following error:

```EXCEPTION_RECORD:  (.exr -1)
ExceptionAddress: 00007ff9c785db00
   ExceptionCode: c0000005 (Access violation)
  ExceptionFlags: 00000000
NumberParameters: 2
   Parameter[0]: 0000000000000008
   Parameter[1]: 00007ff9c785db00
Attempt to execute non-executable address 00007ff9c785db00
```

This same exact error is occurring with both the Indexer and Web Connector.

It's definitely related to MacType - disabling MacType altogether causes the issue to go away - but excluding the Ableton, Ableton Indexer, or Ableton Web Connector exes doesn't seem to fix the issue.

Let me know if there's anything more I can do to help debug this.

**Comment by sammilucia** [comment]
Or it's related to Ableton 😉. Can you try Ableton beta 10.1.b18 please?

**Comment by w-biggs** [comment]
As I noted in https://github.com/snowie2000/mactype/issues/426#issuecomment-481078258, the beta does work. I'm not sure why, though, as the crash doesn't seem to be due to font rendering, but by both processes trying to use the same memory somehow (I think). I figured it was still worth investigating the cause of the issue, since this specific issue may not be limited to Ableton.

**Comment by sammilucia** [comment]
Hmm, Ableton did a lot of work to the font engine in this release,
including masses of bug fixes, and it's a very custom interface. I think if
we haven't seen this problem by now with other software we should close
this as Ableton using invalid memory space. However the final call is up to
@snowie2000

**Comment by snowie2000** [comment]
I’ll try my best to replicate this issue on my system.  I’m quite a bit busy these days so I have to push it to the next month I think.

**Comment by snowie2000** [comment]
I knew I delayed a lot of things especially the development of the Mactype.  It’s just that I’m super busy these months and I’ve got a lot of work from my company and I had to make them come true first.  I’m very sorry for that.

**Comment by sammilucia** [comment]
I would suggest forget the issue with Ableton @snowie2000, and focus on
other parts of MacType. Ableton have fixed the issue and they're about to
launch the new version - so no point.

**Comment by sammilucia** [comment]
Electron apps are now working correctly - though they're rendered with DirectWrite, you are still able to modify their rendering using MacType Beta5's DirectWrite parameters.

**Comment by Mahayu** [comment]
Modern Warfare ( includes earlier games in CoD ) conflicts with MacType 2019.1-beta6.
Relative link:
http://unote.hatenablog.com/entry/20161116/1479326383

**Comment by snowie2000** [comment]
> Modern Warfare ( includes earlier games in CoD ) conflicts with MacType 2019.1-beta6.
> 
> Relative link:
> 
> http://unote.hatenablog.com/entry/20161116/1479326383

Don’t use Mactype on games. Simply unload them. 
For games,
- I don’t have the game your mentioned. 
- Even if I have, my computer is not powerful enough to run those modern games. 
- Even if my computer is powerful enough, I can’t debug those games cuz they all have kinda of anti-debug system built in which prevent regular debugger to attach and debug. And they don’t give you a chance to see the crash log most of the time. 

Conclusion: unload them.

**Comment by duanluan** [comment]
ApiPost：https://www.apipost.cn/download.html

**Comment by snowie2000** [comment]
> ApiPost：https://www.apipost.cn/download.html

So what's the problem with it?
